control_data_collecting_tool/launch/control_data_collecting_tool.launch.py

```diff
@@ -19,31 +19,61 @@
 from ament_index_python.packages import get_package_share_directory
 import launch
 from launch import LaunchService
+from launch.actions import DeclareLaunchArgument
+from launch.substitutions import LaunchConfiguration
 from launch_ros.actions import Node
+import yaml
+
+
+# Load yaml file
+def get_course_name(yaml_file_path):
+    with open(yaml_file_path, "r") as file:
+        data = yaml.safe_load(file)
+
+    # get 'COURSE_NAME'
+    course_name = data.get("/**", {}).get("ros__parameters", {}).get("COURSE_NAME", None)
+    return course_name
 
 
 def generate_launch_description():
+    # Define the argument for map_path
+    DeclareLaunchArgument("map_path", description="Path to the map directory")
+    # Use the map_path argument in parameters
+    map_path = LaunchConfiguration("map_path")
+
+    # Get the path to the common param file
```

Comment:
In https://github.com/autowarefoundation/autoware_tools/pull/156, I modified it so that it works without reading the map_path when the along_road feature is not used.